f8a_version_comparator/item_object.py

```diff
@@ -0,0 +1,219 @@
+            return temp
+        if isinstance(item, IntegerItem):
+            return -1
+        if isinstance(item, StringItem):
+            return self.str_cmp(
+                self.comparable_qualifier(
+                    self.value), self.comparable_qualifier(
+                    item.value))
+        if isinstance(item, ListItem):
+            return -1
+        else:
+            raise ValueError("invalid item" + type(item))
+
+    def to_string(self):
+        """Return value in string form."""
+        return str(self.value)
+
+    @classmethod
+    def is_none(self):
+        """Check if none."""
+        if self.value is release_version_index and self.value is 0:
+            return True
+
+        return False
+
+
+class ListItem(Item):
+    """String Item class for maven version comparator tasks."""
+
+    def __init__(self):
+        """Initialize string value of version."""
+        self.array_list = list()
+
+    def add_item(self, item):
+        """Add item to array list."""
+        arr_list = self.array_list
+        arr_list.append(item)
+
+    def get_list(self):
+        """Get object list items."""
+        return self.array_list
+
+    def normalize(self):
+        """Remove trailing items: 0, "", empty list."""
+        red_list = [0, None, ""]
+        i = len(self.array_list) - 1
+        while(i >= 0):
+            lastItem = self.array_list[i]
+
+            if(not isinstance(lastItem, ListItem)):
+
+                if lastItem.value in red_list:
+                    self.array_list.pop(i)
+                else:
+                    break
+
+            i = i - 1
+
+    def compare_to(self, item):
+        """Compare two maven versions."""
+        if item is None:
+            if len(self.array_list) == 0:
+                return 0
+            first = self.array_list[0]
+            return first.compare_to(None)
+
+        if isinstance(item, IntegerItem):
+            return -1
+        if isinstance(item, StringItem):
+            return 1
+        if isinstance(item, ListItem):
+            left_iter = iter(self.array_list)
+            right_iter = iter(item.get_list())
+
+            while(True):
+                l_obj = next(left_iter, None)
+                r_obj = next(right_iter, None)
+                if l_obj is None and r_obj is None:
+                    break
+                result = 0
+                if l_obj is None:
+                    if r_obj is None:
+                        result = 0
+                    else:
+                        result = -1 * r_obj.compare_to(l_obj)
+                else:
+                    result = l_obj.compare_to(r_obj)
+                if result is not 0:
+                    return result
+
+            return 0
+        else:
+            raise ValueError("invalid item" + type(item))
+
+    def is_none(self):
+        """Check if none."""
+        if len(self.arraylist) is o:
```

Comment:
`o` is not declared anywhere?

We don't call this method anywhere, do we need it?

Reply:
We are not using this method, I can remove it :)